src/internal/m365/onedrive/restore_test.go

```diff
@@ -315,3 +320,74 @@ func (suite *RestoreUnitSuite) TestAugmentRestorePaths_DifferentRestorePath() {
 		})
 	}
 }
+
+func (suite *RestoreUnitSuite) TestRestoreItem_errItemAlreadyExists() {
+	table := []struct {
+		name          string
+		onCollision   control.CollisionPolicy
+		expectErr     func(*testing.T, error)
+		expectSkipped assert.BoolAssertionFunc
+	}{
+		{
+			name:        "skip",
+			onCollision: control.Skip,
+			expectErr: func(t *testing.T, err error) {
+				require.NoError(t, err, clues.ToCore(err))
+			},
+			expectSkipped: assert.True,
+		},
+		{
+			name:        "replace",
+			onCollision: control.Replace,
+			expectErr: func(t *testing.T, err error) {
+				require.ErrorIs(t, err, graph.ErrItemAlreadyExistsConflict, clues.ToCore(err))
+			},
+			expectSkipped: assert.False,
+		},
+	}
+	for _, test := range table {
+		suite.Run(test.name, func() {
+			t := suite.T()
+
+			ctx, flush := tester.NewContext(t)
+			defer flush()
+
+			var (
+				rh = mock.RestoreHandler{
+					PostItemErr: graph.ErrItemAlreadyExistsConflict,
+				}
+				restoreCfg = control.RestoreConfig{
+					OnCollision: test.onCollision,
+				}
+				dpb = odConsts.DriveFolderPrefixBuilder("driveID1")
+			)
+
+			dpp, err := dpb.ToDataLayerOneDrivePath("t", "u", false)
+			require.NoError(t, err)
+
+			dp, err := path.ToDrivePath(dpp)
+			require.NoError(t, err)
+
+			_, skip, err := restoreItem(
+				ctx,
+				rh,
+				restoreCfg,
+				mock.FetchItemByName{
+					Item: &mock.Data{
+						Reader: mock.FileRespReadCloser(mock.DriveFileMetaData),
+					},
+				},
+				version.Backup,
```

Comment:
Should we also be testing lower backup versions?

Reply:
The behavior doesn't change across versions, we should be safe here.